i2pd-tools: init at 2.56.0

MulliganSecurity opened the pull request:
i2pd-tools is a toolsuite that helps working with i2p to generate keys, vanity addresses and so on



## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Sigmanificient commented on maintainers/maintainer-list.nix:
```diff
@@ -16089,6 +16089,12 @@
     githubId = 220262;
     name = "Ion Mudreac";
   };
+  Mulligansecurity = {
```
Make sure to split this addition into a separate commit preceding the package

Sigmanificient commented on pkgs/by-name/i2/i2pd-tools/package.nix:
```diff
@@ -0,0 +1,45 @@
+
+  #tries to access the network during the tests, which fails
+
+  src = pkgs.fetchgit {
+      url = "https://github.com/PurpleI2P/i2pd-tools.git";
+    hash = "sha256-mmCs8AHHKhx1/rDp/Vc1p2W3pufoTa4FcJyJwD919zw=";
+    fetchSubmodules = true;
+  };
```
```suggestion
  src = fetchFromGitHub {
      owner = "PurpleI2P";
      repo = "i2pd-tools";
      hash = "sha256-mmCs8AHHKhx1/rDp/Vc1p2W3pufoTa4FcJyJwD919zw=";
      fetchSubmodules = true;
  };
```
you will need to specify a revision (or a tag) within fetchFromGitHub's attrset, and import `fetchFromGitHub` in the derivation

Sigmanificient commented on pkgs/by-name/i2/i2pd-tools/package.nix:
```diff
@@ -0,0 +1,45 @@
+
+stdenv.mkDerivation rec {
+  pname = "i2pd-tools";
+  version = "unstable";
```
I believe the version is [2.56.0](https://github.com/PurpleI2P/i2pd-tools/commit/33fce4b087d92ee90653460bbe7a07cdc0c7b121)

MulliganSecurity replied:
I understood that as "update for i2pd 2.56.0", should I use the i2pd version number?

FliegendeWurst replied:
There should be a proper version number here. 2.56.0 seems right to me

FliegendeWurst replied:
Although 2.56.0 could just be the version for the used library. In which we set `0-unstable-YYYY-MM-DD` with the date of commit as version

Sigmanificient commented on pkgs/by-name/i2/i2pd-tools/package.nix:
```diff
@@ -0,0 +1,45 @@
+  openssl,
+}:
+
+stdenv.mkDerivation rec {
```
Do not use `rec` here, prefer `(finalAttrs:`

MulliganSecurity replied:
will look that one up as I've never used it.

Sigmanificient commented on pkgs/by-name/i2/i2pd-tools/package.nix:
```diff
@@ -0,0 +1,45 @@
+    fetchSubmodules = true;
+  };
+
+  buildInputs = with pkgs; [ zlib.dev openssl.dev boost.dev ];
```
```suggestion
  buildInputs = [ zlib openssl boost ];
```
You also need to import the library individually

Sigmanificient commented on pkgs/by-name/i2/i2pd-tools/package.nix:
```diff
@@ -0,0 +1,45 @@
+  };
+
+  buildInputs = with pkgs; [ zlib.dev openssl.dev boost.dev ];
+  installPhase = ''
+      mkdir -p $out/bin
+      mv routerinfo $out/bin
+      mv keygen $out/bin
+      mv vain $out/bin
+      mv keyinfo $out/bin
+      mv regaddr $out/bin
+      mv regaddr_3ld $out/bin
+      mv regaddralias $out/bin
+      mv x25519 $out/bin
+      mv famtool $out/bin
+      mv autoconf $out/bin'';
```
```suggestion
  installPhase = ''
    runHook preInstall

    mkdir -p $out/bin
    for bin in \
        routerinfo keygen vain keyinfo regaddr \
        regaddr_3ld regaddralias x25519 famtool autoconf;
    do
      install -Dm755 $bin -t $out/bin
    done

    runHook postInstall
  '';
```

Sigmanificient commented on pkgs/by-name/i2/i2pd-tools/package.nix:
```diff
@@ -0,0 +1,45 @@
+      mv famtool $out/bin
+      mv autoconf $out/bin'';
+
+  meta = {
+    maintainers = with lib.maintainers; [ MulliganSecurity ];
+    mainProgram = "i2pd-tools";
+    license = lib.licenses.bsd3;
+    description = ''
+        Toolsuite to work with keys and eepsites
+    '';
+  };
```
```suggestion
  meta = {
    description = "Toolsuite to work with keys and eepsites";
    license = lib.licenses.bsd3;
    maintainers = with lib.maintainers; [ MulliganSecurity ];
    mainProgram = "i2pd-tools";
  };
```

MulliganSecurity committed:
maintainers list update

MulliganSecurity committed:
i2pd-tools: init at 2.56.0

i2pd-tools is a toolsuite that helps working with i2p to generate
keys, vanity addresses and so on

Sigmanificient commented on pkgs/by-name/i2/i2pd-tools/package.nix:
```diff
@@ -0,0 +1,50 @@
+  meta = {
+    description = "Toolsuite to work with keys and eepsites";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ MulliganSecurity ];
```
Did you added yourself to the maintainers list? The CI show evaluation failing on undefined `MulliganSecurity`

MulliganSecurity replied:
fixed, there was a typo

Sigmanificient commented on pkgs/by-name/i2/i2pd-tools/package.nix:
```diff
@@ -0,0 +1,50 @@
+    maintainers = with lib.maintainers; [ MulliganSecurity ];
+    mainProgram = "i2pd-tools";
+  };
+
```
nits: a newline got here

MulliganSecurity commented:
@FliegendeWurst  can you give me some pointers to bring this one across the finish line?

FliegendeWurst commented on pkgs/by-name/i2/i2pd-tools/package.nix:
```diff
@@ -0,0 +1,49 @@
+  '';
+
+  meta = {
+    description = "Toolsuite to work with keys and eepsites";
```
Please add meta.homepage pointing to the repository
